modopt/opt/proximity.py

```diff
@@ -345,6 +347,77 @@ def _cost_method(self, *args, **kwargs):
         return cost_val
 
 
+class SingularValueThreshold(ProximityParent):
+    r"""Singular Value Threshold operator.
+
+    This is the proximity operator solving:
+    .. math:: arg min \tau||x||_* + 1/2||x||_F
+    """
+
+    def __init__(self, threshold, initial_rank, thresh_type='soft'):
+        self.threshold = threshold
+        self.rank = initial_rank
+        self.threshold_op = SparseThreshold(
+            linear=Identity(),
+            weights=threshold,
+            thresh_type=thresh_type
+        )
```

Comment:
I can see why you did this, but it seems a bit odd. These proximity operators are really designed to be inputs to the algorithms. It would make more sense to me to simply use the `thresh` method directly and handle the `extra_factor` within the `_op_method`. This way you don't need to import `Identity`.